packages/flutter/lib/src/material/material.dart

```diff
@@ -475,3 +498,134 @@ abstract class InkFeature {
   @override
   String toString() => describeIdentity(this);
 }
+
+/// An interpolation between to [ShapeBorder]s.
+///
+/// This class specialises the interpolation of [Tween] to use [ShapeBorder.lerp].
+class ShapeBorderTween extends Tween<ShapeBorder> {
+
+  /// Creates a [ShapeBorder] tween.
+  ///
+  /// the [begin] and [end] properties may be null; see [ShapeBorder.lerp] for
+  /// the null handling semantics.
+  ShapeBorderTween({ShapeBorder begin, ShapeBorder end}): super(begin: begin, end: end);
+
+  /// Returns the value this variable has at the given animation clock value.
```

Comment:
Not sure what you mean by "variable."  Seems more like it should be "[Tween]".

Reply:
done